easybuild/easyblocks/l/lammps.py

```diff
@@ -0,0 +1,353 @@
+                nvcc_wrapper_path = os.path.join(self.start_dir, "lib", "kokkos", "bin", "nvcc_wrapper")
+                self.cfg.update('configopts', '-DKOKKOS_ENABLE_CUDA=yes')
+                self.cfg.update('configopts', '-DCMAKE_CXX_COMPILER="%s"' % nvcc_wrapper_path)
+                self.cfg.update('configopts', '-DCMAKE_CXX_FLAGS="-ccbin $CXX $CXXFLAGS"')
+
+        # CUDA only
+        elif cuda:
+            self.cfg.update('configopts', '-DPKG_GPU=on')
+            self.cfg.update('configopts', '-DGPU_API=cuda')
+
+            self.check_cuda_compute_capabilities(cfg_cuda_cc, ec_cuda_cc, cuda_cc)
+            self.cfg.update('configopts', '-DGPU_ARCH=%s' % self.get_cuda_gpu_arch(cuda_cc))
+
+        return super(EB_LAMMPS, self).configure_step()
+
+    def sanity_check_step(self, *args, **kwargs):
+        check_files = [
+            'atm', 'balance', 'colloid', 'crack', 'dipole', 'friction',
+            'hugoniostat', 'indent', 'melt', 'message', 'min', 'msst',
+            'nemd', 'obstacle', 'pour', 'voronoi',
+        ]
+
+        custom_commands = [
+            # LAMMPS test - you need to call specific test file on path
+            """python -c 'from lammps import lammps; l=lammps(); l.file("%s")'""" %
+            # The path is joined by "build_dir" (start_dir)/examples/filename/in.filename
+            os.path.join(self.start_dir, "examples", "%s" % check_file, "in.%s" % check_file)
+            # And this should be done for every file specified above
+            for check_file in check_files
+        ]
+
+        shlib_ext = get_shared_lib_ext()
+        custom_paths = {
+            'files': [
+                'bin/lmp',
+                'include/lammps/library.h',
+                'lib64/liblammps.%s' % shlib_ext,
+            ],
+            'dirs': [],
+        }
+
+        python = get_software_version('Python')
+        if python:
+            pyshortver = '.'.join(get_software_version('Python').split('.')[:2])
+            pythonpath = os.path.join('lib', 'python%s' % pyshortver, 'site-packages')
+            custom_paths['dirs'].append(pythonpath)
+
+        return super(EB_LAMMPS, self).sanity_check_step(
+            custom_commands=custom_commands,
+            custom_paths=custom_paths,
+        )
+
+    def make_module_extra(self):
+        """Add install path to PYTHONPATH"""
+
+        txt = super(EB_LAMMPS, self).make_module_extra()
+
+        python = get_software_version('Python')
+        if python:
+            pyshortver = '.'.join(get_software_version('Python').split('.')[:2])
+            pythonpath = os.path.join('lib', 'python%s' % pyshortver, 'site-packages')
+            txt += self.module_generator.prepend_paths('PYTHONPATH', [pythonpath])
+
+        txt += self.module_generator.prepend_paths('PYTHONPATH', ["lib64"])
+        txt += self.module_generator.prepend_paths('LD_LIBRARY_PATH', ["lib64"])
+
+        return txt
+
+    def get_cuda_gpu_arch(self, cuda_cc):
```

Comment:
Those are just helper functions and I think having them out of sight (at the bottom) is better, because I expect overridden steps (configure, build, ...) in easyblock the most. So for me it made sense to define helper functions at bottom, and when you encounter the function, you can always search for its definition.

Reply:
OK, I don't care strongly about it, it was just a suggestion.

We usually keep methods "in order".

Maybe you can move these outside of the `class` though, and make them regular functions where you just pass down the stuff they need (like `self.cfg['kokkos_arch']`)